src/client/game.cpp

```diff
@@ -2522,9 +2532,14 @@ void Game::decreaseViewRange()
 void Game::toggleFullViewRange()
 {
 	draw_control->range_all = !draw_control->range_all;
-	if (draw_control->range_all)
-		m_game_ui->showTranslatedStatusText("Enabled unlimited viewing range");
-	else
+	if (draw_control->range_all) {
+		if (sky->getFogDistance() >= 0) {
+			m_game_ui->showTranslatedStatusText("The server has disabled unlimited viewing range");
+			draw_control->range_all = false;
```

Comment:
For consistency with behaviour of view range: Should probably keep the value of the client setting (or user-set value, if not a setting) untouched, and limit it on each use.

Reply:
Agreed. (Ugly to keep more local vs. remote state around.)

Reply:
On second thought... range_all is already a toggle, not retained between restarts, not part of the sky-api itself, and just a debug tool (if you keep it on your game crawls to a halt after a while).
I think it's fine to simply unset it (here and below when the fog distance is set/changed by the server)

Reply:
Ah NM. Makes sense to make it the same.